docs/BUILD.md

```diff
@@ -303,23 +323,61 @@ Note that once you flash `installer.raw` on the installer media, such as USB dri
+1. The Makefile runs `./tools/parse-pkgs.sh images/rootfs-$(HV).yml.in > images/rootfs-$(HV).yml`, i.e. the utility [parse-pkgs.sh](../tools/parse-pkgs.sh), passing it as an input the HV-specific template generated in the previous step `rootfs-$(HV).yml.in`, and saving the output to the final `rootfs-$(HV).yml` file. In addition, the variable `KERNEL_TAG` is passed as an environment variable.
+1. `parse-pkgs.sh` does the following:
+    1. Gets the package tag for each directory in [pkg/](../pkg/) via `linuxkit pkg show-tag ${dir}`, and save it to variable which looks like `<PKGNAME>_TAG`, e.g. `PILLAR_TAG` or `WWAN_TAG`.
+    1. Go through the input file - the HV-specific template - and replace the tags with the appropriate values. This includes the value of `KERNEL_TAG` as passed by the Makefile on calling `parse-pkgs.sh`.
+1. The Makefile generates `rootfs.tar` via `./tools/makerootfs.sh tar -y images/rootfs-$(HV).yml -t path/to/rootfs.tar -a $(ZARCH)`, i.e. it runs [makerootfs.sh](../tools/makerootfs.sh), passing it the following arguments:
+    1. The target format, i.e. `tar`
+    1. `-a $(ZARCH)` - the architecture
+    1. `-t $(PATH)` - path to the target output file
+    1. `-y $(YML)` - path to the yml file, as generated in previous steps
+1. `makerootfs.sh` runs `linuxkit build` to generate the final `rootfs.tar`
+
+The above process creates several challenges, which will, eventually, be cleaned up:
+
+1. The `images/` source directory is unclean, with both version-control-committed code - modifiers `*.yq` and primary template `rootfs.yml.in` - and non-committed code - interim HV-specific template, `rootfs-$(HV).yml.in` and the final output yml file `rootfs-$(HV).yml` - in the same directory.
+2. The same input file, e.g. `rootfs.yml`, appears to be usable with different architectures, but actually is not, as it is architecture-specific.
+3. It is necessary to pre-process the actual source files before generating an image. It is not possible to run `linuxkit build` manually to generate an image. This makes building and debugging individual steps harder.
 
-This is used to replace the tags of various components in the `.yml.in` file with the correct current image name and tag for various packages, including the correct architecture.
+These are all due to constraints within the usage of the `yml` files. If a cleaner solution requires upstreaming into linuxkit, it will be added to the [UPSTREAMING.md](./UPSTREAMING.md) file.
 
-The output `.yml` file is stored in the same directory as the `.yml.in` file, i.e. [images/](../images/).
+### Platform-specific
 
-This creates several challenges, which will, eventually, be cleaned up:
+The current process uses the control knob `PLATFORM` solely as a value in [kernel-version.mk](../kernel-version.mk), in order to set `KERNEL_TAG`, which is passed to [parse-pkgs.sh](../tools/parse-pkgs.sh).
```

Comment:
@rene-zededa don't we use PLATFORM to influence the partition layout, boot loader etc as well? Or is that done when the installer runs?

Reply:
If it changes the build process, we should add it to this document; if it is only part of the installer, it is out of scope for this change.

Reply:
@eriknordmark , yes, we use on i.MX8MP to change the layout partition, it's not only on installer, it also works for live images... so I think the word "solely" must be removed... the rest of the information is correct, we do use PLATFORM as a value in kernel-version.mk to set KERNEL_TAG...

Reply:
@rene I will remove the word solely, but I would like to have a pointer to where else we use it, at the very least. Where else do we consume `PLATFORM` var to determine that layout?

Reply:
I updated it, but not sure I got it correct. I tried following the code, but I might have made errors or missed something. Please have a look.